check_licenses.sh

```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+
+check_file() {
+  if [ -d $file ]
+  then
+    return # This is a directory
+  fi
+  header=$(grep -E "LICENSE" $file)
+  if [ ! "$header" == "" ]
+  then
+    return # There was a LICENSE found.
+  fi
+  echo "$file"
+}
+
+for file in $(git ls-files | grep -E -v "(\.sh|\.md|\.arcs|\.bazel|\.dl|\.facts|\.csv|\.txt|\.yaml|\.authlogic|\.raksha|\.ir)" | grep -E -v "(WORKSPACE|build_defs|Dockerfile|third_party|BUILD|images|gcb|rust)" | grep -E -v "^\.")
```

Comment:
Thanks Mark! I've updated it as a pre-commit hook.
I also:

- Switched to using Pathspecs to make things more managable (and added comments on each group of specs).
- Added support for a 'source' subcommand which checks all 'source' files (not just changed files)
- Added a subcommand for checking 'all' files, regardless of their filename (to use as a reference point)
- Added a help subcommand to make using the script easier (the USAGE string that it echos should be easy to find even if running `./scripts/hooks/pre-commit help` isn't obvious).
- Added licenses to the scripts I've added (amusingly I didn't originally think to include them).
- Moved (non-script) file changes into a second PR which I can land once this is merged.

You're right that this can be bypassed, but the intention is not to enforce anything, but help guide towards best practice.
Looking forward to your review :)

Aside:
It's worth noting that your previous work in #282 (referenced in #285), protects against missing licenses that could impact internal code. Unfortunately I couldn't see a good way to de-duplicate code here, but perhaps that is something your could advise on?

Reply:
I also don't see a good way to dedup the code. The difficulty in sharing code is the big reason why I think it's probably OK to have a commit hook only rather than trying to maintain this code in the presubmit as well. Agreed that guidance, rather than enforcement, is the goal here.